naslib/utils/vis/utils.py

```diff
@@ -0,0 +1,78 @@
+import logging
+import torch
+import numpy as np
+
+import matplotlib.pyplot as plt
+from matplotlib.cm import ScalarMappable
+import seaborn as sns
+
+logger = logging.getLogger(__name__)
+
+def plot_architectural_weights(config, optimizer):
+    all_weights = torch.load(f'{config.save}/arch_weights.pt') # load alphas
+
+    # unpack search space information
+    alpha_dict = {}
+    min_soft, max_soft = np.inf, -np.inf
+    for graph in optimizer.graph._get_child_graphs(single_instances=True):
+        for edge_weights, (u, v, edge_data) in zip(all_weights, graph.edges.data()):
```

Comment:
There's a bug here. `all_weights` is a list of size 28 (in case of nb301), each a tensor of size `(n_steps, n_operations)`. The first 14 are from the normal cells, while the next 14 are from the reduction cells. The loop assigns the same alphas for both normal and reduction cells in line 29. 
![Screenshot 2023-02-01 at 3 47 40 PM](https://user-images.githubusercontent.com/1825878/216076188-9192464a-a396-495d-bb5c-79c9ac4bdb89.png)